src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,36 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: $padding;
+  max-width: calc(7 * (#{$cell-size} + #{$border-width} + #{$gap} + #{$padding}));
+
+  &--start-day-sun {
+    .calendar__cell:nth-child(1) {
+      margin-right: calc((6 * #{$cell-size}) + (6 * #{$border-width}) + (6 * #{$gap}) + (2 * #{$padding}));
+      width: #{$cell-size};
+      height: #{$cell-size};
+    }
+  }
+
+  @for $dayWeek from 28 through 31 {
+    $monthLength: $dayWeek;
+
+    &--month-length-#{$monthLength} {
+      @include generate-calendar-cells($monthLength);
```

Comment:
why do you need it? just create full calendar as default and then you will remove a couple of cell with help of this loop